src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -501,6 +503,645 @@ BOOST_AUTO_TEST_CASE(CharExtractDictionaryEncodingSerialization)
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1, 1, 1, 0, 0 },
+    { 0, 1, 1, 1, 1 },
+    { 1, 1, 0, 1, 0 }
+  };
+
+  CheckMatrices(output, target);
+}
+
+/**
+ * Test the Bag of Words encoding algorithm in case of individual
+ * character encoding, storing resulting in vector<vector<vectorType>>.
+ */
+BOOST_AUTO_TEST_CASE(VectorBagOfWordsEncodingIndividualCharactersTest)
+{
+  std::vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  vector<vector<size_t>> output;
+  BagOfWordsEncoding<CharExtract::TokenType> encoder;
+
+  // Passing a empty string to encode characters
+  encoder.Encode(input, output, CharExtract());
+
+  vector<vector<size_t>> expected = {
+    { 1, 1, 1, 0, 0 },
+    { 0, 1, 1, 1, 1 },
+    { 1, 1, 0, 1, 0 }
+  };
+
+  BOOST_REQUIRE(output == expected);
+}
+
+/**
+ * Test the Tf-Idf Encoding using rawcount type and smoothidf as true,
+ * which is the deafult values used for algorithim.
+ */
+BOOST_AUTO_TEST_CASE(RawCountSmoothIdfEncodingTest)
+{
+  using DictionaryType = StringEncodingDictionary<boost::string_view>;
+
+  arma::mat output;
+  TfIdfEncoding<SplitByAnyOf::TokenType> encoder;
+  SplitByAnyOf tokenizer(" ,.");
+
+  encoder.Encode(stringEncodingInput, output, tokenizer);
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+  arma::mat expected = {
+    {  1.28768207245178, 1.28768207245178, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1, 1.69314718055995, 1.28768207245178,
+       1.28768207245178, 1.28768207245178, 1.69314718055995, 1.69314718055995,
+       1.28768207245178, 1, 1.69314718055995, 1.69314718055995, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0 },
+    {  0, 1.28768207245178, 0, 0, 0, 2, 0, 0, 3.86304621735534,
+       3.86304621735534, 0, 0, 0, 3, 0, 0, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 5.07944154167984, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1.69314718055995, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1.69314718055995, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1.69314718055995, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {  1.28768207245178, 0, 0, 0, 0, 1, 0, 1.28768207245178, 0, 0, 0, 0,
+       1.28768207245178, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 1.69314718055995, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1.69314718055995, 1.69314718055995, 1.69314718055995,
+       1.69314718055995, 1.69314718055995,  1.69314718055995 }
+  };
```

Comment:
Can I ask where these values came from?  Were they hand-calculated, or done using another tool?

Reply:
It's a good question! If I remember right I compared all values with scikit learn. But I think it's a good idea to add the python code analogue to the documentation.

Reply:
Thanks for adding the Python code! :+1: